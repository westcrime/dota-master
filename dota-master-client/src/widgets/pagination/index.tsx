import { Button, Box } from "@mui/material";

interface PaginationProps {
  pageIndex: number;
  hasPreviousPage: boolean;
  hasNextPage: boolean;
  onPageChange: (newPage: number) => void;
}

export const Pagination = ({
  pageIndex,
  hasPreviousPage,
  hasNextPage,
  onPageChange,
}: PaginationProps) => {
  return (
    <Box
      display="flex"
      justifyContent="center"
      alignItems="center"
      gap={2}
      mt={4}
      data-testid="pagination"
      my={4}
    >
      <Button
        variant="contained"
        data-testid="has-previous-page-button"
        onClick={() => onPageChange(pageIndex - 1)}
        disabled={!hasPreviousPage}
        sx={{
          minWidth: 100,
          fontWeight: 'bold',
          textTransform: 'none',
        }}
      >
        Назад
      </Button>
      
      <Button
        variant="contained"
        data-testid="has-next-page-button"
        onClick={() => onPageChange(pageIndex + 1)}
        disabled={!hasNextPage}
        sx={{
          minWidth: 100,
          fontWeight: 'bold',
          textTransform: 'none',
        }}
      >
        Вперед
      </Button>
    </Box>
  );
};