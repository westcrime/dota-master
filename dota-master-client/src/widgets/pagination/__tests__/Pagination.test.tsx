import { render, waitFor } from "@testing-library/react";
import { Pagination } from "@widgets/pagination";

const renderPagination = (
  hasNextPage: boolean,
  hasPreviousPage: boolean,
  pageIndex: number,
  totalPages: number
) =>
  render(
    <Pagination
      hasNextPage={hasNextPage}
      hasPreviousPage={hasPreviousPage}
      onPageChange={jest.fn()}
      pageIndex={pageIndex}
      totalPages={totalPages}
    />
  );

describe("Pagination", () => {
  it("renders", async () => {
    const { queryByTestId } = renderPagination(false, false, 1, 1);

    await waitFor(() => {
      expect(queryByTestId("pagination-text")).toBeInTheDocument();
      expect(queryByTestId("has-next-page-button")).toBeInTheDocument();
      expect(queryByTestId("has-previous-page-button")).toBeInTheDocument();
    });
  });

  it("pageIndex is correct", () => {
    const { getByTestId } = renderPagination(true, true, 2, 3);

    expect(getByTestId("pagination-text")).toHaveTextContent("2 / 3");
  });

  it("buttons are enabled when there are next and previous pages", async () => {
    const { getByTestId } = renderPagination(true, true, 2, 3);

    await waitFor(() => {
      expect(getByTestId("has-next-page-button")).toBeEnabled();
      expect(getByTestId("has-previous-page-button")).toBeEnabled();
    });
  });

  it("buttons are disabled when no next and previous pages", async () => {
    const { getByTestId } = renderPagination(false, false, 2, 3);

    await waitFor(() => {
      expect(getByTestId("has-next-page-button")).toBeDisabled();
      expect(getByTestId("has-previous-page-button")).toBeDisabled();
    });
  });

  it("matches snapshot", () => {
    const { asFragment } = renderPagination(true, true, 2, 3);
    expect(asFragment()).toMatchSnapshot();
  });
});
