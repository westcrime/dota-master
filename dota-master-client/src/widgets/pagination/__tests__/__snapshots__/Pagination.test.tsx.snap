// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mt-4 flex justify-center space-x-4"
    data-testid="pagination"
  >
    <button
      class="px-4 py-2 bg-blue-500 text-white rounded disabled:bg-blue-900 disabled:text-gray-400 disabled:hover:cursor-not-allowed hover:bg-blue-600 focus:outline-none cursor-pointer undefined"
      data-testid="has-previous-page-button"
    >
      Back
    </button>
    <span
      class="px-4 py-2"
      data-testid="pagination-text"
    >
      2 / 3
    </span>
    <button
      class="px-4 py-2 bg-blue-500 text-white rounded disabled:bg-blue-900 disabled:text-gray-400 disabled:hover:cursor-not-allowed hover:bg-blue-600 focus:outline-none cursor-pointer undefined"
      data-testid="has-next-page-button"
    >
      Next
    </button>
  </div>
</DocumentFragment>
`;
